### ISSUE TITLE
blob: Check max blob size and return err if exceeds

### ISSUE THREAD

**Issue by renaynay** [celestiaorg/celestia-node#2495]
Blocked on core/app (cc @rootulp)

**Comment by rootulp** [comment]
See https://github.com/celestiaorg/celestia-app/issues/2108

**Comment by Wondertan** [comment]
I think if that's a stateless check, we should add it to our validation path. Specifically in the blob constructor on our side.

**Comment by rootulp** [comment]
The stateless check depends on a versioned constant ([`SquareSizeUpperBound`](https://github.com/celestiaorg/celestia-app/blob/2a09cfa1b9293380db0254489b0ea8a3e6bb8110/pkg/appconsts/versioned_consts.go#L26-L28)) and a hard-coded constant ([`MaxBlockSizeBytes`](https://github.com/celestiaorg/celestia-core/blob/d02553fdba2720f0314f2f6451a5d50b6755e62c/types/params.go#L14-L15)). We can add it to blob constructor on celestia-node side with the caveat that the versioned constant may change in the future.

**Comment by Wondertan** [comment]
The version, in this case, should be the one binary was built with.

**Comment by rootulp** [comment]
The versioned constant is dependent on the app version which is currently `v1` (see [here](https://github.com/celestiaorg/celestia-app/blob/2a09cfa1b9293380db0254489b0ea8a3e6bb8110/pkg/appconsts/versioned_consts.go#L8)) and likely will be incremented when there's a new major version of celestia-app.

**Comment by vgonkivs** [comment]
Can we simply use `LatestVersion` constant even when It will be v2,v3...?

**Comment by rootulp** [comment]
I think celestia-node can use `LatestVersion` because I would expect celestia-app v1.x.x `LatestVersion` to be v1 and celestia-app v2.x.x's `LatestVersion` to be v2.

**Comment by rootulp** [comment]
Closing as won't do. We've improved the behavior of celestia-app's `CheckTx` in https://github.com/celestiaorg/celestia-app/pull/2202 so txs with a total blob size that is too large will be rejected at the mempool leve.
